Automattic/jetpack
# [WIP] Carousel: Try moving the carousel module to ES6 and bundle with webpack

Fixes #

Note: this is a work in progress exploration into moving the Carousel module's JS to ES6, to hopefully make it easier to read, maintain and test. This an exploration at the moment to see how hard it'll be to migrate the existing code, without actually changing how it functions.

Note: if this approach looks viable, it'll likely need to be re-done pending decisions about whether to refactor the carousel to use Swiper in https://github.com/Automattic/jetpack/pull/20107.

#### Changes proposed in this Pull Request (and to do list items):


* [x] Add webpack config to build the carousel code
* [ ] Add npm scripts to ensure the carousel code is built via watch commands and in the production build
* [x] Create a `src` directory in the carousel module to contain the ES6 source code
* [x] Fix up linting errors with each of the functions, and move to ES6 where appropriate (e.g. swap all the `var`s for `let` and `const`)
* [ ] Skip copying over the `src` directory to the `_inc/build` directory
* [ ] Skip existing minification for the carousel module
* [ ] Remove the old JS code

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Check out this PR
* Start a development build
* Load up your test site with the carousel module switched on, and create a post using the core gallery block
* View the page on the front end of the site
* The carousel should work, and in the HTML of the site, you should see that the new JS bundle is being loaded:
![image](https://user-images.githubusercontent.com/14988353/122698251-46cb9900-d28a-11eb-8cc0-e8f26095e609.png)
